src/Domain/Creators/Interval.hpp

```diff
@@ -76,51 +76,61 @@ class Interval : public DomainCreator<1> {
   };
   template <typename BoundaryConditionsBase>
   struct UpperBoundaryCondition {
-    static std::string name() noexcept { return "UpperBoundary"; }
+    static std::string name() noexcept { return "UpperBoundaryCondition"; }
     static constexpr Options::String help =
         "Options for the boundary condition applied at the upper boundary.";
     using type = std::unique_ptr<BoundaryConditionsBase>;
-    using group = BoundaryConditions;
   };
   template <typename BoundaryConditionsBase>
   struct LowerBoundaryCondition {
-    static std::string name() noexcept { return "LowerBoundary"; }
+    static std::string name() noexcept { return "LowerBoundaryCondition"; }
     static constexpr Options::String help =
         "Options for the boundary condition applied at the lower boundary.";
     using type = std::unique_ptr<BoundaryConditionsBase>;
-    using group = BoundaryConditions;
   };
 
-  using basic_options =
-      tmpl::list<LowerBound, UpperBound, IsPeriodicIn, InitialRefinement,
-                 InitialGridPoints, TimeDependence>;
+  using common_options =
+      tmpl::list<LowerBound, UpperBound, InitialRefinement, InitialGridPoints>;
+
+  using options_periodic = tmpl::list<IsPeriodicIn>;
+  template <typename System>
+  using options_boundary_conditions = tmpl::list<
+      LowerBoundaryCondition<
+          domain::BoundaryConditions::get_boundary_conditions_base<System>>,
+      UpperBoundaryCondition<
+          domain::BoundaryConditions::get_boundary_conditions_base<System>>>;
 
   template <typename Metavariables>
-  using options = tmpl::conditional_t<
-      domain::BoundaryConditions::has_boundary_conditions_base_v<
-          typename Metavariables::system>,
-      tmpl::push_back<
-          basic_options,
-          LowerBoundaryCondition<
-              domain::BoundaryConditions::get_boundary_conditions_base<
-                  typename Metavariables::system>>,
-          UpperBoundaryCondition<
-              domain::BoundaryConditions::get_boundary_conditions_base<
-                  typename Metavariables::system>>>,
-      basic_options>;
+  using options = tmpl::append<
+      common_options,
+      tmpl::conditional_t<
+          domain::BoundaryConditions::has_boundary_conditions_base_v<
+              typename Metavariables::system>,
+          options_boundary_conditions<typename Metavariables::system>,
+          options_periodic>,
+      // Need TimeDependence at the end to not break the already broken python
+      // bindings. This is so that the constructors can default them to nullptr.
```

Comment:
`git apply` this, then you can pass `nullptr`s to the constructor in the order you want:

```diff
diff --git a/src/Domain/Creators/Python/Interval.cpp b/src/Domain/Creators/Python/Interval.cpp
index e3027692f..0c7a80af4 100644
--- a/src/Domain/Creators/Python/Interval.cpp
+++ b/src/Domain/Creators/Python/Interval.cpp
@@ -17,9 +17,18 @@ namespace creators {
 namespace py_bindings {
 void bind_interval(py::module& m) {  // NOLINT
   py::class_<Interval, DomainCreator<1>>(m, "Interval")
-      .def(py::init<std::array<double, 1>, std::array<double, 1>,
-                    std::array<bool, 1>, std::array<size_t, 1>,
-                    std::array<size_t, 1>>(),
+      .def(py::init(
+               [](std::array<double, 1> lower_x, std::array<double, 1> upper_x,
+                  std::array<bool, 1> is_periodic_in_x,
+                  std::array<size_t, 1> initial_refinement_level_x,
+                  std::array<size_t, 1> initial_number_of_grid_points_in_x) {
+                 return Interval{std::move(lower_x),
+                                 std::move(upper_x),
+                                 std::move(is_periodic_in_x),
+                                 std::move(initial_refinement_level_x),
+                                 std::move(initial_number_of_grid_points_in_x),
+                                 nullptr};
+               }),
            py::arg("lower_x"), py::arg("upper_x"), py::arg("is_periodic_in_x"),
            py::arg("initial_refinement_level_x"),
            py::arg("initial_number_of_grid_points_in_x"));
```